Mathlib/LinearAlgebra/Dimension/Basic.lean

```diff
@@ -63,9 +63,22 @@ protected irreducible_def Module.rank : Cardinal :=
 theorem rank_le_card : Module.rank R M ≤ #M :=
   (Module.rank_def _ _).trans_le (ciSup_le' fun _ ↦ mk_set_le _)
 
-lemma nonempty_linearIndependent_set : Nonempty {s : Set M // LinearIndepOn R id s } :=
+instance nonempty_linearIndependent_set : Nonempty {s : Set M // LinearIndepOn R id s } :=
   ⟨⟨∅, linearIndepOn_empty _ _⟩⟩
 
+variable [Nontrivial R]
+
+/-- See `rank_subsingleton` that assumes `Subsingleton R` instead. -/
+@[simp, nontriviality] theorem rank_subsingleton' [Subsingleton M] : Module.rank R M = 0 := by
+  rw [Module.rank, ← bot_eq_zero, eq_bot_iff]
+  exact ciSup_le fun s ↦ by simp [(linearIndependent_subsingleton_iff _).mp s.2]
+
+@[simp]
+theorem rank_punit : Module.rank R PUnit = 0 := rank_subsingleton' _ _
+
+@[simp]
+theorem rank_bot : Module.rank R (⊥ : Submodule R M) = 0 := rank_subsingleton' _ _
```

Comment:
Is there a reason why these are moved? The idea is to put every thing mentioning a finite number in `Finite.lean`.

Reply:
This file imports only 813 files but the original file imports 1085. I think it's better to make such basic results available earlier.

Reply:
Perhaps you could make a `Zero.lean` and `One.lean` and move things there instead?

Reply:
Maybe I'll create `Subsingleton.lean` and move `rank_subsingleton` there as well.